Message formatting in "Wait For Elements State" uses the given timeout even when it is None

**Description**
When "Wait For Elements State" keyword is called with timeout argument set as None, and message argument containing formatter for timeout (i.e. {timeout}), the message for raised AssertionError is formatted with the timeout being None, instead of using the default timeout, which is used for the actual timeout for wait.

**To Reproduce**
```
from Browser import Browser
browser = Browser()
browser.new_browser()
browser.set_browser_timeout('20s')
browser.new_page('https://robotframework.org/')
browser.wait_for_elements_state('//not-found-locator', timeout=None, message='Did not find {selector} in {timeout}')
```

**Actual behavior**
"AssertionError: Did not find //not-found-locator in None"

**Expected behavior**
"AssertionError: Did not find //not-found-locator in 20s" (or 20000ms)

**Versions**
robotframework: 4.0.3
robotframework-assertion-engine: 0.0.6
robotframework-browser: 5.1.1
Python: 3.8.10

--- a/Browser/keywords/waiter.py
+++ b/Browser/keywords/waiter.py
@@ -54,6 +54,7 @@ class Waiter(LibraryComponent):
         [https://docs.python.org/3/library/stdtypes.html#str.format|format] options.
         The `{function}` formatter is same ``state`` argument value.
         """
+        timeout_as_str = self.millisecs_to_timestr(self.get_timeout(timeout))
         funct = {
             ElementState.enabled: "e => !e.disabled",
             ElementState.disabled: "e => e.disabled",
@@ -84,9 +85,10 @@ class Waiter(LibraryComponent):
                         logger.debug(f"Suppress error: {error}")
                     else:
                         if message:
-                            timeout = self.millisecs_to_timestr(self.get_timeout(timeout))
                             message = message.format(
-                                selector=selector, function=state, timeout=timeout
+                                selector=selector,
+                                function=state,
+                                timeout=timeout_as_str,
                             )
                             raise AssertionError(message)
                         raise
@@ -147,6 +149,7 @@ class Waiter(LibraryComponent):
         | Click         \\#progress_bar
         | Wait For      ${promise}
         """
+        timeout_as_str = self.millisecs_to_timestr(self.get_timeout(timeout))
         end = float(
             self.convert_timeout(timeout, False) if timeout else self.timeout / 1000
         )
@@ -159,9 +162,8 @@ class Waiter(LibraryComponent):
                     logger.debug(f"Suppress {error}")
                 else:
                     if message:
-                        timeout = self.millisecs_to_timestr(self.get_timeout(timeout))
                         message = message.format(
-                            selector=selector, function=function, timeout=timeout
+                            selector=selector, function=function, timeout=timeout_as_str
                         )
                         raise AssertionError(message)
                     raise

--- a/Browser/keywords/waiter.py
+++ b/Browser/keywords/waiter.py
@@ -84,6 +84,7 @@ class Waiter(LibraryComponent):
                         logger.debug(f"Suppress error: {error}")
                     else:
                         if message:
+                            timeout = self.millisecs_to_timestr(self.get_timeout(timeout))
                             message = message.format(
                                 selector=selector, function=state, timeout=timeout
                             )
@@ -158,6 +159,7 @@ class Waiter(LibraryComponent):
                     logger.debug(f"Suppress {error}")
                 else:
                     if message:
+                        timeout = self.millisecs_to_timestr(self.get_timeout(timeout))
                         message = message.format(
                             selector=selector, function=function, timeout=timeout
                         )

--- a/atest/test/03_Waiting/wait_for_element_state.robot
+++ b/atest/test/03_Waiting/wait_for_element_state.robot
@@ -76,11 +76,18 @@ Wait For Elements State fails On Too Short Timeout
     ...    STARTS:TimeoutError: page.waitForFunction: Timeout 300ms exceeded.
     ...    Wait For Elements State    \#victim    unchecked    300ms
 
+Wait For Elements State fails On Too Short Global Timeout
+    ${timeout} =    Set Browser Timeout    0.3 s
+    Run Keyword and Expect Error
+    ...    Custom Error #submit, ElementState.hidden and 300 milliseconds
+    ...    Wait For Elements State    \#submit    hidden    ${None}    Custom Error {selector}, {function} and {timeout}
+    [Teardown]    Set Browser Timeout    ${timeout}
+
 Wait For Elements State Fails On Too Short Timeout Custom Error With Formatting
     Select Options By    \#dropdown    value    unchecked
     Click    \#submit    noWaitAfter=True
     Run Keyword and Expect Error
-    ...    Custom Error #victim, e => !e.checked and 0:00:00.300000
+    ...    Custom Error #victim, e => !e.checked and 300 milliseconds
     ...    Wait For Elements State    \#victim    unchecked    300ms    Custom Error {selector}, {function} and {timeout}
 
 Wait For Elements State Fails On Too Short Timeout Custom Error
@@ -94,7 +101,7 @@ Wait For Elements State Fails On Too Short Timeout Custom Error And Hidden
     Select Options By    \#dropdown    value    unchecked
     Click    \#submit    noWaitAfter=True
     Run Keyword and Expect Error
-    ...    Custom Error #victim, ElementState.hidden and 0:00:00.300000
+    ...    Custom Error #victim, ElementState.hidden and 300 milliseconds
     ...    Wait For Elements State    \#victim    hidden    300ms    Custom Error {selector}, {function} and {timeout}
 
 Wait For Elements State to hide with Promise

